docs/source/traits_user_manual/advanced.rst

```diff
@@ -38,15 +38,23 @@ attribute, with a name based on the name of the trait attribute:
 
 .. method:: _name_default()
 
-This method initializes the *name* trait attribute, returning its initial value.
-The method overrides any default value specified in the trait definition. The
-initializer is called when:
+This method returns the default value for the *name* trait attribute and it
+overrides any default value specified in the trait definition.
+
+Similar to static default values, default values defined dynamically are
+considered to have existed **prior to** setting object state during
```

Comment:
perhaps "... should be thought of as existing..." instead of "...are considered to have existed...".  Probably need to fix up the tense of the rest of the sentence if you make that change.